bulk-model-sync-mps/src/main/kotlin/org/modelix/mps/model/sync/bulk/MPSBulkSynchronizer.kt

```diff
@@ -141,6 +151,118 @@ object MPSBulkSynchronizer {
         }
     }
 
+    @OptIn(ExperimentalSerializationApi::class)
+    @JvmStatic
+    fun importRepositoryFromModelServer() {
+        val repository = getRepository()
+
+        val includedModuleNames = parseRawPropertySet(System.getProperty("modelix.mps.model.sync.bulk.input.modules"))
+        val includedModulePrefixes = parseRawPropertySet(System.getProperty("modelix.mps.model.sync.bulk.input.modules.prefixes"))
+        val continueOnError = System.getProperty("modelix.mps.model.sync.bulk.input.continueOnError", "false").toBoolean()
+
+        val modelServerUrl = System.getProperty("modelix.mps.model.sync.bulk.server.url")
+        val repositoryId = RepositoryId(requireNotNull(System.getProperty("modelix.mps.model.sync.bulk.server.repository")) { "modelix.mps.model.sync.bulk.server.repository not specified" })
+        val branchName = System.getProperty("modelix.mps.model.sync.bulk.server.branch")
+        val branchRef = repositoryId.getBranchReference(branchName)
+        val versionHash: String? = System.getProperty("modelix.mps.model.sync.bulk.server.version.hash")
+        val baseVersionHash: String? = System.getProperty("modelix.mps.model.sync.bulk.server.version.base.hash")
+        val client = ModelClientV2.builder().url(modelServerUrl).build()
+        val version = runBlocking {
+            if (versionHash == null) client.lazyLoadVersion(branchRef) else client.lazyLoadVersion(repositoryId, versionHash)
+        }
+        val baseVersion = baseVersionHash?.let {
+            runBlocking {
+                client.lazyLoadVersion(repositoryId, it)
+            }
+        }
+        println("Loading version ${version.getContentHash()}")
+
+        val access = repository.modelAccess
+        access.executeCommandInEDT {
+            if (baseVersion != null) {
```

Comment:
What about an early exit avoid the ongoing nesting of the happy path such as:

```suggestion
            if (baseVersion == null) {
                return
            }
```

Also, if `baseVersion` is null, this should probably error out?